### PR TITLE
Fix: Re-create browsers after killing all sessions on preboot 

### PR DESCRIPTION
**Pull request by andyMrtnzP**
Fixes  https://github.com/browserless/chrome/issues/3030

### CODE REVIEW

File: src/chrome-helper.ts [hunk 1]
```diff
@@ -601,13 +605,15 @@ export const launchChrome = async (
...
     throw e;
   });
-
   const { webSocketDebuggerUrl: browserWSEndpoint } = await fetchJson(
     `http://127.0.0.1:${port}/json/version`,
   ).catch((e) => {
     browserServer.close();
     throw e;
   });
+  console.log(
```
Comment (joelgriffith) [inline src/chrome-helper.ts:614]: Might want to `debug` or something here?

File: src/puppeteer-provider.ts [hunk 2]
```diff
@@ -576,7 +584,7 @@ export class PuppeteerProvider {
 
       jobdebug(`${job.id}: Browser cleanup complete.`);
 
-      if (this.config.prebootChrome && browser._prebooted) {
```
Comment (joelgriffith) [inline src/puppeteer-provider.ts:579]: I'm pretty sure this is necessary... but it's not hitting me as to why

**Comment by joelgriffith** [comment]
How about we just do this:

**Comment by andyMrtnzP** [comment]
> In `route.ts`. Might still have to change the puppeteer-provider to clean out it's internal swarm cache of `chromeSwarm`. But might be a bit cleaner instead of passing instances around?

Yes, we ended up needing this change. And yes, it is way cleaner

File: src/chrome-helper.ts [hunk 3]
```diff
@@ -759,10 +759,6 @@ export const getProtocolJSON = async () => {
 
 export const killAll = async () => {
   await Promise.all(runningBrowsers.map((browser) => closeBrowser(browser)));
-
-  runningBrowsers = [];
```
Comment (joelgriffith) [inline src/chrome-helper.ts:763]: This because `closeBrowser` takes care of removing them? Just wanted to double check

File: src/routes.ts [hunk 4]
```diff
@@ -295,6 +295,7 @@ export const getRoutes = ({
   if (!disabledFeatures.includes(Features.KILL_ENDPOINT)) {
     router.get('/kill/all', async (_req, res) => {
       await chromeHelper.killAll();
+      puppeteerProvider.setSwarm(await puppeteerProvider.start());
```
Comment (joelgriffith) [inline src/routes.ts:298]: Sorry for the churn on this one. Why don't we just make `puppeteerProvider.start` "reset" the `chromeSwarm` array anytime it's called? Maybe even rename it so it's more obvious that it's rotating the chrome instances it caches. Just trying to make it as idiomatic as possible.

**Review by joelgriffith — CHANGES_REQUESTED** [review]
One open question + a final thought. Thanks for seeing this one through

**Comment by joelgriffith** [comment]
https://github.com/browserless/chrome/pull/3073/files#diff-7d02faf941f1b3ac4166544452aaafa8ebc68af58f356a84dec8d57c2af0a81eR80 is still manually pushing in chrome instances as they're created

**Review by joelgriffith — APPROVED** [review]

**Pull request merged**